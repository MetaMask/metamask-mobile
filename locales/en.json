{
	"autocomplete": {
		"placeholder": "Enter website address"
	},
	"navigation": {
		"back": "Back",
		"close": "Close",
		"cancel": "Cancel",
		"info": "Info"
	},
	"onboarding": {
		"title": "Welcome to MetaMask",
		"subtitle": "A browser that connects you to applications on the decentralized web",
		"import_wallet_button": "Import a wallet",
		"start_exploring_now": "Start exploring",
		"login": "Login"
	},
	"create_wallet": {
		"title": "Creating your new wallet",
		"subtitle": "Sit tight while we do some hard math! You will use your wallet to login & interact securely with decentralized applications"
	},
	"import_wallet": {
		"title": "Import Wallet",
		"sync_help": "Sync your wallet with the extension",
		"sync_help_step_one": "1. Open the extension",
		"sync_help_step_two": "2. Go to settings",
		"sync_help_step_three": "3. Click on \"Sync with Mobile\"",
		"sync_help_step_four": "4. Scan the QR Code",
		"sync_from_browser_extension_button": "Start Syncing",
		"or": "OR",
		"import_from_seed_button": "Import using Seed Phrase"
	},
	"login": {
		"title": "Welcome Back!",
		"password": "Password",
		"login_button": "LOG IN",
		"go_back": "Go Back",
		"invalid_password": "Invalid password"
	},
	"choose_password": {
		"title": "Create password",
		"subtitle": "You'll use this password to unlock your MetaMask wallet only on this device.",
		"password": "New Password",
		"confirm_password": "Confirm Password",
		"create_button": "NEXT",
		"import_with_seed_phrase": "Import with seed phrase",
		"password_length_error": "The password needs to be at least 8 chars long",
		"password_dont_match": "Passwords don't match",
		"password_strength": "Password strength:",
		"strength_weak": "Weak",
		"strength_good": "Good",
		"strength_strong": "Strong",
		"show": "Show",
		"hide": "Hide",
		"must_be_at_least": "Must be at least {{number}} characters",
		"remember_me": "Remember me",
		"security_alert_title": "Security Alert",
		"security_alert_message": "In order to proceed, you need to turn Passcode on or any biometrics authentication method supported in your device (FaceID, TouchID or Fingerprint)"
	},
	"import_from_seed": {
		"title": "Import from seed",
		"seed_phrase_placeholder": "Enter your seed phrase here",
		"new_password": "New Password (min. 8 characters)",
		"confirm_password": "Confirm Password",
		"import_button": "IMPORT",
		"cancel_button": "Cancel",
		"password_length_error": "The password needs to be at least 8 chars long",
		"password_dont_match": "Passwords don't match",
		"seed_word_count_error": "The seed needs to be 12 words"
	},
	"bottom_tab_bar": {
		"dapps": "√êApps",
		"wallet": "Wallet",
		"transfer": "Transfer"
	},
	"drawer": {
		"send_button": "Send",
		"receive_button": "Receive",
		"coming_soon": "Coming soon...",
		"wallet": "Wallet",
		"transaction_history": "Transaction History",
		"submit_feedback": "Send Feedback",
		"submit_feedback_message": "Choose the type of feedback to send.",
		"submit_bug": "BUG REPORT",
		"submit_general_feedback": "GENERAL",
		"share_address": "Share my Public Address",
		"view_in_etherscan": "View on Etherscan",
		"browser": "Browser",
		"settings": "Settings",
		"help": "Get help",
		"logout": "Log Out",
		"logout_title": "Do you really want to log out?",
		"logout_ok": "YES",
		"logout_cancel": "NO",
		"feedback": "Feedback",
		"metamask_support": "MetaMask Support"
	},
	"send": {
		"title": "Send"
	},
	"receive": {
		"title": "Receive"
	},
	"wallet": {
		"title": "Wallet",
		"tokens": "TOKENS",
		"collectible": "Collectible",
		"collectibles": "COLLECTIBLES",
		"transactions": "TRANSACTIONS",
		"no_collectibles": "You don't have any collectibles!",
		"add_tokens": "ADD TOKENS",
		"no_tokens": "You don't have any tokens!",
		"add_collectibles": "ADD COLLECTIBLES",
		"no_transactions": "You have no transactions!",
		"send_button": "Send",
		"deposit_button": "Deposit",
		"copy_address": "Copy",
		"remove_token_title": "Do you want to remove this token?",
		"remove_collectible_title": "Do you want to remove this collectible?",
		"remove": "Remove",
		"cancel": "Cancel"
	},
	"add_asset": {
		"title": "Add Asset",
		"search_token": "SEARCH",
		"custom_token": "CUSTOM TOKEN",
		"tokens": {
			"cancel_add_token": "CANCEL",
			"add_token": "ADD TOKEN"
		},
		"collectibles": {
			"cancel_add_collectible": "CANCEL",
			"add_collectible": "ADD"
		}
	},
	"token": {
		"token_symbol": "Token Symbol",
		"token_address": "Token Address",
		"token_precision": "Token of Precision",
		"search_tokens_placeholder": "Search Tokens",
		"address_cant_be_empty": "Token address can't be empty.",
		"address_must_be_valid": "Token address must be a valid address.",
		"symbol_cant_be_empty": "Token symbol can't be empty.",
		"decimals_cant_be_empty": "Token decimals can't be empty.",
		"select_token": "Select Token"
	},
	"collectible": {
		"collectible_address": "Address",
		"collectible_type": "Type",
		"collectible_token_id": "ID",
		"collectible_description": "Description",
		"address_must_be_valid": "Collectible address must be a valid address.",
		"address_must_be_smart_contract": "Personal address detected. Input the collectible contract address.",
		"address_cant_be_empty": "Collectible address can't be empty.",
		"token_id_cant_be_empty": "Collectible identifier can't be empty.",
		"ownership_error_title": "Ooops! Something happened.",
		"ownership_error": "You are not the owner of this collectible, so you can't add it.",
		"powered_by_opensea": "Powered by"
	},
	"transfer": {
		"title": "Transfer",
		"send": "SEND",
		"receive": "RECEIVE"
	},
	"accounts": {
		"create_new_account": "Create New Account",
		"import_account": "Import an Account",
		"imported": "IMPORTED"
	},
	"app_settings": {
		"title": "Settings",
		"current_conversion": "Base Currency",
		"current_language": "Current Language",
		"search_engine": "Search Engine",
		"new_RPC_URL": "New RPC Network",
		"state_logs": "State Logs",
		"auto_lock": "Auto-lock",
		"auto_lock_desc": "Choose the amount of time before the application automatically locks.",
		"state_logs_desc": "This will help MetaMask debug any issue you might encounter. Please send it via email to mobile@metamask.io",
		"autolock_immediately": "Immediately",
		"autolock_never": "Never",
		"autolock_after": "After {{time}} seconds",
		"reveal_seed_words": "Reveal Seed Words",
		"reset_account": "Reset Account",
		"state_logs_button": "DOWNLOAD STATE LOGS",
		"reveal_seed_words_button": "REVEAL SEED WORDS",
		"reset_account_button": "RESET ACCOUNT",
		"reset_account_confirm_button": "RESET",
		"reset_account_cancel_button": "NEVERMIND",
		"reset_account_modal_title": "Reset Account?",
		"clear_approvals_modal_title": "Clear Approval Data?",
		"clear_approvals_modal_message": "All dapps will need to request access to view account information again.",
		"clear_browser_history_modal_title": "Clear Browser History?",
		"clear_browser_history_modal_message": "We are about to remove all your browser history. Are you sure?",
		"reset_account_modal_message": "Resetting your account will clear your transaction history.",
		"save_rpc_url": "SAVE",
		"invalid_rpc_prefix": "URIs require the appropriate HTTPS prefix",
		"invalid_rpc_url": "Invalid RPC URL",
		"sync_with_extension": "Sync with Metamask Extension",
		"sync_with_extension_desc": "This will import all extension accounts to this device.",
		"sync": "SYNC",
		"clear_approved_dapps": "CLEAR APPROVED DAPPS",
		"clear_browser_history": "CLEAR BROWSER HISTORY",
		"clear_approve_dapps_desc": "Clear approved dapps",
		"clear_browser_history_desc": "Clear browser history",
		"clear": "CLEAR",
		"privacy_mode": "Privacy mode",
		"privacy_mode_desc": "Websites must request access to view your account information.",
		"show_hex_data": "Show Hex Data",
		"show_hex_data_desc": "Select this to show the hex data field on the send screen.",
		"general_title": "General",
		"general_desc": "Currency conversion, primary currency, language",
		"advanced_title": "Advanced",
		"advanced_desc": "Access developer features, Reset Account, setup testnets and custom RPC",
		"security_title": "Security & Privacy",
		"security_desc": "Privacy settings, private key and wallet seed phrase",
		"info_title": "About MetaMask",
		"legal_title": "Legal",
		"conversion_title": "Currency conversion",
		"conversion_desc": "Display fiat values in using a specific currency throughout the application.",
		"language_desc": "Translate the application to a different supported language.",
		"engine_desc": "Change the default search engine used when entering search terms in the URL bar.",
		"reset_desc": "This action will clear your transaction history. This data might not be retrievable.",
		"rpc_desc": "Use a custom RPC-capable network via URL instead of one of the provided networks.",
		"hex_desc": "Select this to show the hex data field on the send screen.",
		"clear_privacy_title": "Clear privacy data",
		"clear_privacy_desc": "Clear privacy data so all websites must request access to view account information again.",
		"clear_history_desc": "Choose this option to clear your entire browsing history."
	},
	"app_information": {
		"title": "Information",
		"links": "Links",
		"privacy_policy": "Privacy Policy",
		"terms_of_use": "Terms of use",
		"attributions": "Attributions",
		"support_center": "Visit our Support Center",
		"web_site": "Visit our Website",
		"email_us": "Email Us!"
	},
	"reveal_credential": {
		"seed_phrase_title": "Reveal Seed Phrase",
		"private_key_title": "Show Private Key",
		"show_private_key": "Show Private Key",
		"private_key_title_for_account": "Show Private Key for {{accountName}}",
		"cancel": "Cancel",
		"confirm": "Next",
		"seed_phrase_explanation": "If you ever change browsers or move computers, you will need this seed phrase to access your accounts. Save them somewhere safe and secret.",
		"private_key_explanation": "Save it somewhere safe and secret.",
		"private_key_warning": "This is the private key for the current selected account: {{accountName}}. Never disclose this key. Anyone with your private keys can take steal any assets held in your account.",
		"seed_phrase_warning_explanation": "DO NOT share this phrase with anyone! These words can be used to steal all your accounts.",
		"private_key_warning_explanation": "Never disclose this key. Anyone with your private keys can take steal any assets held in your account.",
		"seed_phrase": "Your seed phrase",
		"private_key": "Your private key",
		"copy_to_clipboard": "Copy to clipboard",
		"enter_password": "Enter password to continue",
		"seed_phrase_copied": "Seed phrase copied to clipboard",
		"private_key_copied": "Private key copied to clipboard",
		"warning_incorrect_password": "Incorrect password",
		"unknown_error": "Couldn't unlock your account. Please try again.",
		"seed_warning": "This is your wallet's 12 word phrase. This is a very sensitive phrase. DO NOT SHARE THIS WITH ANYONE."
	},
	"asset_overview": {
		"send_button": "Send",
		"receive_button": "RECEIVE",
		"add_collectible_button": "Add",
		"info": "Info",
		"description": "Description",
		"totalSupply": "Total Supply",
		"address": "Address"
	},
	"account_details": {
		"title": "Account Details",
		"share_account": "Share",
		"view_account": "View account on Etherscan",
		"show_private_key": "Show private key",
		"account_copied_to_clipboard": "Public address copied to clipboard",
		"share_public_key": "Sharing my public key: "
	},
	"qr_scanner": {
		"invalid_qr_code_title": "Invalid QR Code",
		"invalid_qr_code_message": "The QR code that you are trying to scan it is not valid.",
		"allow_camera_dialog_title": "Allow camera access",
		"allow_camera_dialog_message": "We need your permission to scan QR codes",
		"scanning": "scanning..."
	},
	"action_view": {
		"cancel": "Cancel",
		"confirm": "Confirm"
	},
	"transaction": {
		"alert": "ALERT",
		"amount": "Amount",
		"confirm": "Confirm",
		"reject": "Reject",
		"edit": "Edit",
		"from": "From",
		"gas_fee": "Transaction Fee",
		"gas_fee_fast": "FAST",
		"gas_fee_average": "AVERAGE",
		"gas_fee_slow": "SLOW",
		"hex_data": "Hex Data",
		"review_details": "DETAILS",
		"review_data": "DATA",
		"review_function_type": "FUNCTION TYPE",
		"review_hex_data": "HEX DATA",
		"insufficient": "Insufficient funds",
		"invalid_address": "Invalid address",
		"invalid_amount": "Invalid amount",
		"invalid_gas": "Invalid gas amount",
		"invalid_gas_price": "Invalid gas price",
		"invalid_collectible_ownership": "You don't own this collectible",
		"max": "Max",
		"recipient_address": "Recipient Address",
		"required": "Required",
		"to": "To",
		"total": "Total",
		"loading": "Loading...",
		"conversion_not_available": "Conversion rate not available",
		"value_not_available": "Not Available",
		"rate_not_available": "Conversion not available",
		"optional": "Optional",
		"no_address_for_ens": "No address for ENS name"
	},
	"custom_gas": {
		"advanced_options": "Advanced Options",
		"hide_advanced_options": "Hide Advanced Options",
		"gas_limit": "Gas Limit:",
		"gas_price": "Gas Price: (GWEI)",
		"warning_gas_limit": "Gas limit must be greater than 20999 and less than 7920027"
	},
	"home_page": {
		"lets_get_started": "Let's get started!",
		"web3_awaits": "The decentralized web awaits",
		"search_placeholder": "Search or type URL",
		"backup_alert_title": "Protect your funds",
		"backup_alert_message": "Tap to save your seed phrase",
		"bookmarks": "Bookmarks",
		"no_bookmarks": "You have no favorites Dapps yet",
		"remove_bookmark_title": "Do you want to remove this site from your favorites?"
	},
	"browser": {
		"title": "Browser",
		"reload": "Reload",
		"share": "Share",
		"bookmark": "Bookmark",
		"add_to_favorites": "Add to Favorites",
		"error": "Ooops!",
		"bookmark_already_exists": "You already bookmarked this url",
		"cancel": "Cancel",
		"go_back": "Go back",
		"go_forward": "Go forward",
		"close": "Close",
		"open_in_browser": "Open in browser",
		"change_url": "Change url",
		"dapp_browser": "DAPP BROWSER",
		"dapp_browser_message": "MetaMask is your wallet and browser for the decentralized web. Have a look around!",
		"featured_dapps": "FEATURED DAPPS",
		"my_favorites": "MY FAVORITES",
		"search": "Search or type a URL",
		"welcome": "Welcome!",
		"remove": "Remove"
	},
	"add_favorite": {
		"title": "Add Favorite",
		"title_label": "Name",
		"url_label": "Url",
		"add_button": "Add",
		"cancel_button": "Cancel"
	},
	"approval": {
		"title": "Confirm Transaction"
	},
	"transactions": {
		"tx_review_confirm": "CONFIRM",
		"tx_review_transfer": "TRANSFER",
		"tx_review_contract_deployment": "CONTRACT DEPLOYMENT",
		"tx_review_transfer_from": "TRANSFER FROM",
		"tx_review_unknown": "UNKNOWN METHOD",
		"sent_ether": "Sent Ether",
		"self_sent_ether": "Sent Yourself Ether",
		"received_ether": "Received Ether",
		"sent_tokens": "Sent Tokens",
		"sent_collectible": "Sent Collectible",
		"sent": "Sent",
		"contract_deploy": "Contract Deployment",
		"to_contract": "New Contract",
		"smart_contract_interaction": "Smart Contract Interaction",
		"hash": "Hash",
		"from": "From",
		"to": "To",
		"details": "Details",
		"amount": "Amount",
		"gas_limit": "Gas Limit (Units)",
		"gas_price": "Gas Price (GWEI)",
		"total": "Total",
		"view_on_etherscan": "VIEW ON ETHERSCAN",
		"hash_copied_to_clipboard": "Transaction hash copied to clipboard"
	},
	"transaction_submitted": {
		"title": "Transaction Submitted",
		"your_tx_hash_is": "Your transaction hash is:",
		"view_on_etherscan": "VIEW ON ETHERSCAN"
	},
	"sync_with_extension": {
		"title": "Sync data from Extension",
		"label": "Open the MetaMask extension in your desktop browser, then go to settings, click on \"Sync with Mobile\" and then enter your password to see the QR Code",
		"button_continue": "Continue",
		"please_wait": "Please wait while we sync your data...",
		"error_title": "Bummer!",
		"error_message": "Something went wrong... Please try again",
		"warning_title": "Warning!",
		"warning_message": "This will override all your current accounts and transaction history in your mobile phone. Are you sure?",
		"warning_ok_button": "YES",
		"warning_cancel_button": "NO"
	},
	"sync_with_extension_success": {
		"title": "Sync completed",
		"sync_complete": "We have synchronized all the information from your MetaMask extension and now you're ready to go!",
		"button_continue": "Continue"
	},
	"networks": {
		"title": "Networks",
		"other_networks": "Other Networks",
		"close": "Close",
		"status_ok": "All Systems Operational",
		"status_not_ok": "The network is having some issues"
	},
	"select": {
		"cancel": "Cancel",
		"done": "Done"
	},
	"signature_request": {
		"title": "Signature Request",
		"cancel": "CANCEL",
		"sign": "SIGN",
		"sign_requested": "Your signature is being requested",
		"signing": "You are signing:",
		"account_title": "Account:",
		"balance_title": "Balance:",
		"message": "Message:"
	},
	"unit": {
		"eth": "ETH",
		"negative": "-",
		"divisor": "/",
		"token_id": "#",
		"colon": ":"
	},
	"biometrics": {
		"enable_touchid": "Sign In with Touch ID?",
		"enable_faceid": "Sign In with Face ID?",
		"enable_fingerprint": "Sign In with Fingerprint?"
	},
	"authentication": {
		"auth_prompt_title": "Authentication required",
		"auth_prompt_desc": "Please authenticate in order to use MetaMask",
		"fingerprint_prompt_title": "Authentication required",
		"fingerprint_prompt_desc": "Use your fingerprint to unlock MetaMask",
		"fingerprint_prompt_cancel": "Cancel"
	},
	"accountApproval": {
		"title": "CONNECT REQUEST",
		"action": "would like to",
		"connect": "CONNECT",
		"cancel": "CANCEL",
		"permission": "View your",
		"address": "public address",
		"warning": "By clicking connect, you allow this dapp to view your public address. This is an important security step to protect your data from potential phishing risks."
	},
	"import_private_key": {
		"title": "Import Account",
		"description_one": "Imported accounts are viewable in your wallet but are not recoverable with your MetaMask seed phrase.",
		"learn_more_here": "Learn more about imported accounts here.",
		"subtitle": "Paste your private key string",
		"cta_text": "IMPORT",
		"example": "e.g. 3a1076bf45ab87712ad64ccb3b10217737f7faacbf2872e88fdd9a537d8fe266",
		"error_title": "Ooops! something went wrong...",
		"error_message": "We couldn't import that private key. Please make sure you entered it correctly.",
		"error_empty_message": "You need to enter your private key."
	},
	"import_private_key_success": {
		"title": "Account succesfully imported!",
		"description_one": "You'll be now able to view your account in MetaMask."
	},
	"first_incoming_transaction": {
		"title": "{{asset}} was deposited in your account",
		"amount": "Amount:",
		"account": "Account:",
		"from": "From:",
		"cta_text": "OK"
	},
	"secure_your_wallet": {
		"title": "Secure your wallet",
		"step_1": "Step 1:",
		"step_1_description": "Create password",
		"step_2": "Step 2:",
		"step_2_description": "Save wallet seed phrase",
		"info_text_1": "Take a few moments to finish setting up your MetaMask wallet.",
		"info_text_2": "This will ensure only you can access your funds and will let you recover your wallet if you lose your device",
		"cta_text": "CREATE PASSWORD"
	},
	"account_backup_step_1": {
		"cancel_backup_title": "Cancel Backup",
		"cancel_backup_message": "We highly recommend you save your seed phrase in order to restore your wallet.",
		"cancel_backup_ok": "Yes, I'll take the risk",
		"cancel_backup_no": "No, back up Seed Phrase",
		"skip": "Skip",
		"important": "IMPORTANT!",
		"title": "Save your wallet's seed phrase",
		"highly_recommended": "HIGHLY RECOMMENDED",
		"info_text_1": "Your wallet's seed phrase is:",
		"info_text_2_1": "A ",
		"info_text_2_2": "secret",
		"info_text_2_3": " code that you do NOT share with anyone.",
		"info_text_3_1": "used to ",
		"info_text_3_2": "recover your wallet",
		"info_text_3_3": " in case you lose your device, forget your password, have to re-install the MetaMask app or want to access your wallet on another device.",
		"cta_text": "YEAH, LET'S SAVE IT",
		"learn_more": "Learn More"
	},
	"account_backup_step_2": {
		"cancel_backup_title": "Cancel Backup",
		"cancel_backup_message": "We highly recommend you save your seed phrase in order to restore your wallet.",
		"cancel_backup_ok": "Yes, I'll take the risk",
		"cancel_backup_no": "No, back up Seed Phrase",
		"title": "Grab a pen and paper",
		"info": "Next step is to write your seed phrase down.",
		"info_2_1": "You will be asked to ",
		"info_2_2": "re-enter",
		"info_2_3": " it for confirmation",
		"cta_text": "OK"
	},
	"account_backup_step_3": {
		"cancel_backup_title": "Cancel Backup",
		"cancel_backup_message": "We highly recommend you save your seed phrase in order to restore your wallet.",
		"cancel_backup_ok": "Yes, I'll take the risk",
		"cancel_backup_no": "No, back up Seed Phrase",
		"title": "Is anyone watching?",
		"info_text": "Make sure no other human or robot is watching your screen. If your seed phrase is copied, it can be used on other devices to steal your funds",
		"cta_text": "NO ONE'S WATCHING ME"
	},
	"account_backup_step_4": {
		"cancel_backup_title": "Cancel Backup",
		"cancel_backup_message": "We highly recommend you save your seed phrase in order to restore your wallet.",
		"cancel_backup_ok": "Yes, I'll take the risk",
		"cancel_backup_no": "No, back up Seed Phrase",
		"back": "Back",
		"title": "Your seed phrase",
		"info_text_1": "Carefully write down these words on paper. Their order matters.",
		"info_text_2": "You'll be asked to re-enter it on the next screen",
		"cta_text": "I'VE COPIED THE PHRASE",
		"confirm_password": "Confirm your password",
		"before_continiuing": "Before continuing we need you to confirm your password",
		"confirm": "CONFIRM"
	},
	"account_backup_step_5": {
		"error_title": "Oops!",
		"error_message": "The order of the words is incorrect. Please make sure you wrote it down correctly and go to the previous screen if it is necessary",
		"back": "Back",
		"title": "Confirm seed phrase",
		"info_text": "Insert each word in the order it was presented to you on the previous screen.",
		"cta_text": "CONFIRM PHRASE",
		"modal_title": "Seed phrase confirmed!",
		"modal_text": "This was to ensure you follow this security measure",
		"modal_button": "NEXT"
	},
	"account_backup_step_6": {
		"title": "Security Tips",
		"info_text": "MetaMask cannot recover your seed phrase if you lose it",
		"tip_1": "Keep multiple backups of your seed phrase",
		"tip_2": "Store the phrase in a trusted password manager and paper backups in a safe place",
		"tip_3": "Never share this phrase with anyone",
		"disclaimer": "* You can find your seedphrase if you go to ",
		"disclaimer_bold": "Settings > Security & Privacy",
		"cta_text": "GOT IT!",
		"modal_title": "Congratulations!",
		"modal_text": "You are all backed up and ready to go!",
		"modal_button": "DONE",
		"copy_seed_phrase": "COPY SEED PHRASE TO CLIPBOARD"
	},
	"phishing": {
		"ethereum_phishing_detection": "Ethereum Phishing Detection",
		"ethereum_phishing_detector": "Ethereum Phishing Detector",
		"intro": " is currently on the MetaMask domain warning list. This means that based on information available to us, MetaMask believes this domain could currently compromise your security and, as an added safety feature, MetaMask has restricted access to the site. To override this, please read the rest of this warning for instructions on how to continue at your own risk.",
		"reasons": "There are many reasons sites can appear on our warning list, and our warning list compiles from other widely used industry lists. Such reasons can include known fraud or security risks, such as domains that test positive on the ",
		"list_content": "Domains on these warning lists may include outright malicious websites and legitimate websites that have been compromised by a malicious actor.",
		"to_read_more": "To read more about this site ",
		"review_on_etherscam": "please review the domain on Etherscam.",
		"warning": "Note that this warning list is compiled on a voluntary basis. This list may be inaccurate or incomplete. Just because a domain does not appear on this list is not an implicit guarantee of that domain's safety. As always, your transactions are your own responsibility. If you wish to interact with any domain on our warning list, you can do so by ",
		"continue_on_your_own": "continuing at your own risk.",
		"file_an_issue_intro": "If you think this domain is incorrectly flagged or if a blocked legitimate website has resolved its security issues, ",
		"file_an_issue": "please file an issue.",
		"back_to_safety": "Back to safety"
	},
	"notifications": {
		"pending_title": "Transaction submitted!",
		"success_title": "Transaction #{{nonce}} Complete!",
		"error_title": "Oops, something went wrong :/",
		"received_title": "You received {{amount}} {{assetType}}!",
		"pending_message": "Waiting for confirmation...",
		"error_message": "Tap to view this transaction",
		"success_message": "Tap to view this transaction",
		"received_message": "Tap to view this transaction"
	},
	"secure_your_wallet_modal": {
		"title": "Secure your wallet",
		"body": "Now that you have value in your wallet, create password & save your wallet's seed phrase ",
		"required": "(required)",
		"button": "PROCEED NOW"
	}
}
