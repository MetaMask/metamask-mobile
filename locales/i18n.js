import ReactNative from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import I18n from 'react-native-i18n';
import { LANGUAGE } from '../app/constants/storage';

// Import all locales
import en from './en.json';
import es from './es.json';

// Should the app fallback to English if user locale doesn't exists
I18n.fallbacks = true;
I18n.defaultLocale = 'en';
// Define the supported translations
I18n.translations = {
	en,
	es
};
// If language selected get locale
getUserPreferableLocale();

const currentLocale = I18n.currentLocale();

// Is it a RTL language?
export const isRTL = currentLocale.indexOf('he') === 0 || currentLocale.indexOf('ar') === 0;

// Set locale
export async function setLocale(locale) {
	I18n.locale = locale;
	await AsyncStorage.setItem(LANGUAGE, locale);
}

// Get languages
export function getLanguages() {
	return { en: 'English', es: 'Spanish' };
}

// Allow RTL alignment in RTL languages
ReactNative.I18nManager.allowRTL(isRTL);

// The method we'll use instead of a regular string
export function strings(name, params = {}) {
	return I18n.t(name, params);
}

// Allow persist locale after app closed
async function getUserPreferableLocale() {
	const locale = await AsyncStorage.getItem(LANGUAGE);
	if (locale) {
		I18n.locale = locale;
	}
}

export default I18n;
