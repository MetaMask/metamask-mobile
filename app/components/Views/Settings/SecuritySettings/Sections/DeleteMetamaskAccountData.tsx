import React from 'react';
import SettingsButtonSection from '../../../../UI/SettingsButtonSection';
import { strings } from '../../../../../../locales/i18n';
import Text, {
  TextColor,
  TextVariant,
} from '../../../../../component-library/components/Texts/Text';
import Button, {
  ButtonSize,
  ButtonVariants,
} from '../../../../../component-library/components/Buttons/Button';
import { Linking } from 'react-native';
import {
  CONSENSYS_PRIVACY_POLICY,
  DELETE_ACCOUNT_DATA_FORM_URL,
} from '../../../../../constants/urls';
import { selectSeedlessOnboardingLoginFlow } from '../../../../../selectors/seedlessOnboardingController';
import { useSelector } from 'react-redux';

const DeleteMetamaskAccountData = () => {
  const openPrivacyPolicy = () => Linking.openURL(CONSENSYS_PRIVACY_POLICY);
  const openDeleteAccountDataLink = () =>
    Linking.openURL(DELETE_ACCOUNT_DATA_FORM_URL);
  const isSocialLogin = useSelector(selectSeedlessOnboardingLoginFlow);

  if (!isSocialLogin) {
    return null;
  }

  return (
    <SettingsButtonSection
      testID="delete-metamask-account-data-button"
      needsModal={false}
      sectionTitle={strings('app_settings.delete_metamask_account_data_title')}
      sectionButtonText={strings(
        'app_settings.delete_metamask_account_data_button',
      )}
      onPress={openDeleteAccountDataLink}
      descriptionText={
        <>
          <Text variant={TextVariant.BodyMD} color={TextColor.Alternative}>
            {strings('app_settings.delete_metamask_account_data_description')}
          </Text>{' '}
          <Button
            variant={ButtonVariants.Link}
            size={ButtonSize.Auto}
            onPress={openPrivacyPolicy}
            label={strings(
              'app_settings.delete_metamask_account_data_description_privacy_policy',
            )}
          />
        </>
      }
    />
  );
};

export default React.memo(DeleteMetamaskAccountData);
