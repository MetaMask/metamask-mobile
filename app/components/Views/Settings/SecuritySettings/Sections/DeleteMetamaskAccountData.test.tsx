import React from 'react';
import DeleteMetamaskAccountData from './DeleteMetamaskAccountData';
import renderWithProvider from '../../../../../util/test/renderWithProvider';
import { strings } from '../../../../../../locales/i18n';
import { Linking } from 'react-native';
import { act, fireEvent } from '@testing-library/react-native';
import {
  CONSENSYS_PRIVACY_POLICY,
  DELETE_ACCOUNT_DATA_FORM_URL,
} from '../../../../../constants/urls';

describe('DeleteMetamaskAccountData', () => {
  const initialState = {
    engine: {
      backgroundState: {
        SeedlessOnboardingController: {
          vault: 'test',
        },
      },
    },
  };

  it('renders match snapshot', () => {
    const { toJSON } = renderWithProvider(<DeleteMetamaskAccountData />, {
      state: initialState,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders nothing if not social login', () => {
    const { toJSON } = renderWithProvider(<DeleteMetamaskAccountData />, {
      state: {
        engine: {
          backgroundState: {
            SeedlessOnboardingController: {
              vault: undefined,
            },
          },
        },
      },
    });
    expect(toJSON()).toBeNull();
  });

  it('renders the settings section if social login flow', async () => {
    const { getByText } = renderWithProvider(<DeleteMetamaskAccountData />, {
      state: initialState,
    });
    expect(
      getByText(strings('app_settings.delete_metamask_account_data_title')),
    ).toBeOnTheScreen();
  });

  it('opens the delete account data link when the button is pressed', async () => {
    const { getByText } = renderWithProvider(<DeleteMetamaskAccountData />, {
      state: initialState,
    });
    const button = getByText(
      strings('app_settings.delete_metamask_account_data_button'),
    );
    expect(button).toBeOnTheScreen();

    await act(async () => {
      fireEvent.press(button);
    });

    expect(Linking.openURL).toHaveBeenCalledWith(DELETE_ACCOUNT_DATA_FORM_URL);
  });

  it('opens the privacy policy link when the button is pressed', async () => {
    const { getByText } = renderWithProvider(<DeleteMetamaskAccountData />, {
      state: initialState,
    });

    const button = getByText(
      strings(
        'app_settings.delete_metamask_account_data_description_privacy_policy',
      ),
    );

    await act(async () => {
      fireEvent.press(button);
    });

    expect(Linking.openURL).toHaveBeenCalledWith(CONSENSYS_PRIVACY_POLICY);
  });
});
