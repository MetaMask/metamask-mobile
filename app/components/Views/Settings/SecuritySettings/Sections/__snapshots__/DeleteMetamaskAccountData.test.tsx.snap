// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMetamaskAccountData renders match snapshot 1`] = `
<View
  style={
    {
      "marginTop": 32,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#121314",
        "fontFamily": "Geist Medium",
        "fontSize": 18,
        "letterSpacing": 0,
        "lineHeight": 24,
      }
    }
  >
    Delete MetaMask account data
  </Text>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#686e7d",
        "fontFamily": "Geist Regular",
        "fontSize": 16,
        "letterSpacing": 0,
        "lineHeight": 24,
        "marginTop": 8,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#686e7d",
          "fontFamily": "Geist Regular",
          "fontSize": 16,
          "letterSpacing": 0,
          "lineHeight": 24,
        }
      }
    >
      This will delete historical Metamask account data associated with your use across different devices. This process may take up to 30 days.
    </Text>
     
    <Text
      accessibilityRole="link"
      accessible={true}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "backgroundColor": "transparent",
          "color": "#121314",
          "fontFamily": "Geist Regular",
          "fontSize": 16,
          "letterSpacing": 0,
          "lineHeight": 24,
        }
      }
      suppressHighlighting={true}
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#4459ff",
            "fontFamily": "Geist Medium",
            "fontSize": 16,
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        View our Privacy policy.
      </Text>
    </Text>
  </Text>
  <View
    style={
      {
        "marginTop": 16,
      }
    }
  >
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
      testID="delete-metamask-account-data-button"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#121314",
            "fontFamily": "Geist Medium",
            "fontSize": 16,
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        Delete account data
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
